### DISCUSSION TITLE
ORDER BY is not implemented for window functions

### DISCUSSION THREAD

**Discussion by soerenwolfers** [duckdb/duckdb#10966] [General]
```sql
FROM
(VALUES (0, 1), (1, 0)) t(a, b)
SELECT
first(a ORDER BY b) OVER ()
```
prints
```
ORDER BY is not implemented for window functions
```
https://github.com/duckdb/duckdb/issues/4545 links to a paper that talks about how this can be done and is closed as completed. Was this part intentionally omitted? Are there plans to add it later?

**Comment by RobinDarrah** [comment]
@soerenwolfers 

```SQL
FROM
(VALUES (0, 1), (1, 0)) t(a, b)
SELECT
first(a) OVER (ORDER BY b)
```
The above works and is how window functions are implemented in the majority of databases I've encountered - or were you expecting a different result if the order by is included in the first() clause?

**Reply by soerenwolfers** [reply]
Yes, they do different things: the ORDER BY X in the window defines the frame, i.e. the bag of rows to feed into the aggregate function, which in this case would be all rows whose X value is smaller than that of the current row;  the ORDER BY Y  in the aggregate function is just another parameter to that function, which in this case tells it how to pick a single row out of the rows in the bag it's fed, namely by picking the row with the smallest Y (out of all rows that have smaller X than the current row).

**Comment by soerenwolfers** [comment]
@hawkfish mentioned in a comment on https://github.com/duckdb/duckdb/issues/11419 that this is on his radar for 1.0.
